powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ApplicationConfigServiceBehavior.java

```diff
@@ -0,0 +1,176 @@
+            item.setValues(List.copyOf(config.getValues()));
+            responseConfigs.add(item);
+        });
+        response.setApplicationConfigs(responseConfigs);
+        return response;
+    }
+
+    /**
+     * Create an application configuration.
+     * @param request Request for creating application configuration
+     * @return Create application configuration response.
+     * @throws GenericServiceException In case of a business logic error.
+     */
+    public CreateApplicationConfigResponse createApplicationConfig(final CreateApplicationConfigRequest request) throws GenericServiceException {
+        final String applicationId = request.getApplicationId();
+        final String key = request.getKey();
+        final List<String> values = request.getValues();
+        if (applicationId == null) {
+            logger.warn("Invalid application ID in createApplicationConfig");
+            // Rollback is not required, error occurs before writing to database
+            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
+        }
+        if (key == null) {
+            logger.warn("Invalid key in createApplicationConfig");
+            // Rollback is not required, error occurs before writing to database
+            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
+        }
+        final ApplicationRepository appRepository = repositoryCatalogue.getApplicationRepository();
+        final Optional<ApplicationEntity> appOptional = appRepository.findById(applicationId);
+        if (appOptional.isEmpty()) {
+            logger.info("Application not found, application ID: {}", applicationId);
+            // Rollback is not required, error occurs before writing to database
+            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_APPLICATION);
+        }
+        final ApplicationEntity appEntity = appOptional.get();
+        final ApplicationConfigRepository configRepository = repositoryCatalogue.getApplicationConfigRepository();
+        final List<ApplicationConfigEntity> configs = configRepository.findByApplicationId(applicationId);
+        final Optional<ApplicationConfigEntity> matchedConfig = configs.stream()
+                .filter(config -> config.getKey().equals(key))
+                .findFirst();
+        matchedConfig.ifPresentOrElse(config -> {
+            config.setValues(values);
+            configRepository.save(config);
+        }, () -> {
+            final ApplicationConfigEntity config = new ApplicationConfigEntity();
+            config.setApplication(appEntity);
+            config.setKey(key);
+            config.setValues(values);
+            configRepository.save(config);
+        });
+        final CreateApplicationConfigResponse response = new CreateApplicationConfigResponse();
+        response.setApplicationId(applicationId);
+        response.setKey(key);
+        response.setValues(List.copyOf(values));
```

Comment:
🤔 Is it really NPE safe? I afraid that it is possible to send explicit `null` in the JSON.

Reply:
I can add the null check.